arch/WASM/WASMDisassembler.c

```diff
@@ -321,7 +321,7 @@ static uint64_t get_varuint64(const uint8_t *code, size_t code_len, size_t *leng
 			return 0;
 		}
 
-		data = data + ((code[i] & 0x7f) << (i * 7));
+		data = data + (((uint64_t) code[i] & 0x7f) << (i * 7));
```

Comment:
this should fit in a separate pull req, so i can merge it independently?

Reply:
indeed, do you want me to submit a separate pull request ?

Reply:
yes, please

Reply:
or if @SpikeI agree with every changes here, i can merge all of them, without needing you to create a new PR.